src/auth/auth.controller.ts

```diff
@@ -15,61 +26,54 @@ export class AuthController {
 
     @Post('register')
     async registerUser(
-      @Body(ValidationPipe) credentials: RegisterDto): Promise<User> {
-      return await this.authService.registerUser(credentials);
+        @Body(ValidationPipe) credentials: RegisterDto
+    ): Promise<AccessTokenDto> {
+        return await this.authService.registerUser(credentials);
     }
-  
+
     @Post('login')
```

Comment:
Why is login a guard? Does passport require it to be one? It's a one time logic why not place directly into the service with dto validation. Let's talk about that.

Reply:
While you are right that this is one time logic and this does not HAVE to be a Guard, I still would highly recommend to leave it as is. The reason being that this way all endpoints in the controller behave the same as in they get the user data for all called service methods from the Guard.
So not necessary but in my opinion easier to understand in this scenario

Reply:
serialization will solve this

Reply:
> While you are right that this is one time logic and this does not HAVE to be a Guard, I still would highly recommend to leave it as is. The reason being that this way all endpoints in the controller behave the same as in they get the user data for all called service methods from the Guard. So not necessary but in my opinion easier to understand in this scenario

Do you know what the passport constructor does?
```
constructor(private authService: AuthService) {
        super({
            usernameField: 'email',
            role: 'role'
        });
    }
```

Reply:
> serialization will solve this

How?

Reply:
> > serialization will solve this
> 
> How?

idk man ask past me